ocaml/idl/datamodel_tracing.ml

```diff
@@ -0,0 +1,184 @@
+(*
+ * Copyright (C) 2023 Cloud Software Group
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published
+ * by the Free Software Foundation; version 2.1 only. with the special
+ * exception on linking described in file LICENSE.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *)
+
+open Datamodel_types
+open Datamodel_common
+open Datamodel_roles
+
+let create =
+  call ~name:"create" ~in_oss_since:None ~lifecycle:[]
+    ~doc:"Set the fields of a TracerProvider in the Tracing Library"
+    ~params:
+      [
+        (String, "name_label", "The TracerProvider")
+      ; ( Set (Ref _host)
+        , "hosts"
+        , "Hosts the TracerProvider is to be created on"
+        )
+      ; (Bool, "enabled", "The enabled of the TracerProvider")
+      ; (Map (String, String), "tags", "The tags of the TracerProvider")
+      ; (Set String, "endpoints", "The endpoints of the TracerProvider")
+      ; (Set String, "components", "The componenets of the TracerProvider")
+      ; (Set String, "filters", "The filters of the TracerProvider")
+      ; (Set String, "processors", "The processors of the TracerProvider")
```

Comment:
More verbose descriptions, explaining their meaning, would work better for the fields as we don't want to be cryptic
```suggestion
      ; (Bool, "enabled", "Whether the tracer provider is gathering traces or not")
      ; (Map (String, String), "tags", "The user tags that identify the tracer provider")
      ; (Set String, "endpoints", "The file/HTTP endpoints the provider sends traces to")
      ; (Set String, "components", "???")
      ; (Set String, "filters", "???")
      ; (Set String, "processors", "???")
```

Reply:
Resolved [2eab70b](https://github.com/xapi-project/xen-api/pull/4966/commits/2eab70b666d23cd0747c1fc823356685c1b86e93)